### PR TITLE
Fix DLNA PlayTo encoding issue

### PR DESCRIPTION
**Pull request by BaronGreenback**
**Fix 1:** Didl metadata wasn't being populated (all the time???) as flush was absent.

**Fix 2:** Exception caused by non-string GUID being parsed. 
(Raised by the DLNA eventing when the TV was already playing alternative content that contained the word "audios" in the url)

Additional errors still exist in metadata parsing around url encoding, which results in xml parsing errors. This is addressed in the plugin.

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

**Comment by BaronGreenback** [comment]
Further issue https://stackoverflow.com/questions/28422609/how-to-send-setavtransporturi-using-upnp-c/29129958

Some devices don't like empty CurrentURIMetaData fields. Seems my Hisense tv is one of them!

**Comment by Keridos** [comment]
I am gonna test this change tomorrow on my TV.

**Comment by BaronGreenback** [comment]
Good luck! Just working with my hisense TV to get the plugin working - hence the fixes. Planning to move on to Sharp and then BubbleUPnP. Just hoping fixing for one - doesn't break it for another.

**Review by Ullmie02 — APPROVED** [review]

File: Emby.Dlna/Didl/DidlBuilder.cs [hunk 1]
```diff
@@ -113,6 +113,8 @@ public string GetItemDidl(BaseItem item, User user, BaseItem context, string dev
 
                     writer.WriteFullEndElement();
                     // writer.WriteEndDocument();
+
+                    writer.Flush();
```
Comment (cvium) [inline Emby.Dlna/Didl/DidlBuilder.cs:117]: You sure this is necessary? When XmlWriter is disposed it should call Close on itself and Flush.

File: Emby.Dlna/PlayTo/TransportCommands.cs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 using System.Collections.Generic;
 using System.Globalization;
 using System.Linq;
+using System.Security;
```
Comment (cvium) [inline Emby.Dlna/PlayTo/TransportCommands.cs:7]: Is this needed?

Comment (BaronGreenback) [reply Emby.Dlna/PlayTo/TransportCommands.cs:7]: leftover - removed

File: Emby.Dlna/Didl/DidlBuilder.cs (hunk 1 above)
Comment (BaronGreenback) [reply Emby.Dlna/Didl/DidlBuilder.cs]: Backported from the plugin where the using is no longer in brackets (as suggested by compiler).

Have changed to a warning comment - as following the analyser's suggestion breaks it.

**Pull request merged**